src/cssm.pyx

```diff
@@ -5270,4 +4870,208 @@ def ddm_flexbound_tradeoff(np.ndarray[float, ndim = 1] vh,
+                'n_trials': n_trials,
+                'simulator': 'exgauss',
+                'possible_choices': [-1, 1],
+                'delta_t': delta_t, 
+                'max_t': max_t,
+            }
+        }
+    elif return_option == 'minimal':
+        return {'rts': rts, 'choices': choices, 'metadata': {'simulator': 'exgauss', 'n_samples': n_samples, 'n_trials': n_trials, 'possible_choices': [-1, 1]}}
+    else:
+        raise ValueError("return_option must be 'full' or 'minimal'") 
+# -----------------------------------------------------------------------------------------------
+
+
+# Simulate (rt, choice) tuples from: SHIFTED WALD ------------------------------------
+# @cythonboundscheck(False)
+# @cythonwraparound(False)
+def shifted_wald(np.ndarray[float, ndim = 1] gamma,
+                           np.ndarray[float, ndim = 1] alpha, 
+                           np.ndarray[float, ndim = 1] theta, 
+                           np.ndarray[float, ndim = 1] p, # choice probability 
+                           np.ndarray[float, ndim = 1] s, # noise sigma 
+                           float delta_t = 0.001,
+                           float max_t = 20,
+                           int n_samples = 20000,
+                           int n_trials = 1,
+                           random_state = None,
+                           return_option = 'full', 
+                           smooth_unif = False,
+                           **kwargs):
+    """ Uhhhhhh thing for shifted wald """ 
+
+    set_seed(random_state)
+    
+    cdef float[:] gamma_view = gamma 
+    cdef float[:] alpha_view = alpha 
+    cdef float[:] theta_view = theta 
+    cdef float[:] p_view = p
+    cdef float[:] s_view = s
+
+    traj = np.zeros((int(max_t / delta_t) + 1, 1), dtype=DTYPE)
+    traj[:, :] = -999
+    cdef float[:, :] traj_view = traj
+
+    rts = np.zeros((n_samples, n_trials, 1), dtype = DTYPE)
+    choices = np.zeros((n_samples, n_trials, 1), dtype = np.intc)
+    cdef float[:, :, :] rts_view = rts
+    cdef int[:, :, :] choices_view = choices
+
+    cdef float delta_t_sqrt = sqrt(delta_t)
+    cdef float y, t_particle, smooth_u, sqrt_st
+
+    cdef Py_ssize_t n, ix, k
+    cdef int m = 0 
+    cdef int num_draws = int(max_t / delta_t) + 1
+    cdef float[:] gaussian_values = draw_gaussian(num_draws)
+
+
+    for k in range(n_trials): 
+        sqrt_st = delta_t_sqrt * s_view[k]
+
+        for n in range(n_samples): 
+            y = 0.0 
+            t_particle = 0.0 
+            ix = 0 
+
+            if n == 0:
+                if k == 0:
+                    traj_view[0, 0] = y
+
+            # Random choice depending on p 
+            random_val = rand() / float(RAND_MAX) 
+            if random_val <= p_view[k]: 
+                choices_view[n, k, 0] = 1
+            else:
+                choices_view[n, k, 0] = -1
+
+            # Random walk 
+            while (y < alpha_view[k]) and (t_particle <= max_t):
```

Comment:
this one (and the exgauss for that matter), can have two versions. 
One where the choice is determined by a "race", and one where we have the independent bernoulli choice process. 

At the simulator level this is easy enough, at the likelihood level we can first just do the latter, and talk once more to make sure defining the former is clear + analytically tractable enough.